devlib/target.py

```diff
@@ -3138,6 +3138,416 @@ def _resolve_paths(self):
+                name = self.execute('sw_vers -productName').strip()
+                return {
+                    'name': name,
+                    'version': version,
+                    'build': build,
+                }
+            except TargetStableError:
+                return {}
+        else:
+            # Use parent implementation for Linux and other systems
+            return super(LocalTarget, self).os_version
+
+    @property
+    @memoized
+    def system_id(self):
+        """Get a unique system identifier"""
+        if self._is_macos():
+            try:
+                # Use hardware UUID as unique identifier
+                uuid = self.execute('system_profiler SPHardwareDataType | grep "Hardware UUID" | awk \'{print $3}\'').strip()
+                kernel = self.execute('uname -r').strip()
+                return f'{uuid}/{kernel}'
+            except TargetStableError:
+                return 'unknown'
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).system_id
+
+    @property
+    @memoized
+    def cpuinfo(self):
+        """Get CPU information, using platform-appropriate method"""
+        if self._is_macos():
+            # On macOS, /proc/cpuinfo doesn't exist, use sysctl
+            return self._get_macos_cpuinfo()
+        else:
+            # Use parent implementation for Linux systems
+            return super(LocalTarget, self).cpuinfo
+
+    def _get_macos_cpuinfo(self):
+        """Generate /proc/cpuinfo-like output on macOS using sysctl"""
+        # Get CPU count
+        try:
+            ncpu = int(self.execute('sysctl -n hw.ncpu').strip())
+        except (TargetStableError, ValueError):
+            ncpu = 1
+        
+        # Get CPU brand string
+        try:
+            brand_string = self.execute('sysctl -n machdep.cpu.brand_string').strip()
+        except TargetStableError:
+            brand_string = 'Unknown CPU'
+        
+        # Get CPU family, model, etc. (if available)
+        cpu_family = ''
+        cpu_model = ''
+        try:
+            cpu_family = self.execute('sysctl -n machdep.cpu.family').strip()
+        except TargetStableError:
+            pass
+        
+        try:
+            cpu_model = self.execute('sysctl -n machdep.cpu.model').strip()
+        except TargetStableError:
+            pass
+        
+        # Generate /proc/cpuinfo-like format
+        cpuinfo_text = []
+        for i in range(ncpu):
+            section = [
+                f'processor\t: {i}',
+                f'model name\t: {brand_string}',
+            ]
+            if cpu_family:
+                section.append(f'cpu family\t: {cpu_family}')
+            if cpu_model:
+                section.append(f'model\t\t: {cpu_model}')
+            
+            cpuinfo_text.append('\n'.join(section))
+        
+        return Cpuinfo('\n\n'.join(cpuinfo_text))
```

Comment:
A new class that allows passing the various components explicitly is needed here. This would be a base class of the current `Cpuinfo`, and the latter could simply parse the text format in its `__init__`. That base class would be used directly here, or we can make a subclass that does essentially nothing to keep the base private.

The only issue with that scheme is that `isinstance(x, Cpuinfo)` would now be inaccurate, but I don't think anyone relies on that as there is only a single Cpuinfo class currently.